File: code/chapter4/linear_regression_boston.py
from sklearn.datasets import load_boston
from linear_regression import LR
boston = load_boston()
X, y = boston.data, boston.target
# first, let's run our own linear regression
lr = LR()
lr.fit(X, y)
print(lr.coef)
print(lr.predict(X)[:5])

# now, use sklearn's linear regression model
from sklearn.linear_model import LinearRegression
reg = LinearRegression().fit(X, y)
print(reg.coef_)
print(reg.predict(X)[:5])

File: code/chapter4/linear_regression_boston.R
source('linear_regression.R')

library(MASS) # load Boston data from this package

lr = LR$new()
# -i means excluding the ith column from the data.frame
lr$fit(data.matrix(Boston[,-ncol(Boston)]),Boston$medv)
print(lr$coef)
# let's make prediction on the same data
pred=lr$predict(data.matrix(Boston[,-ncol(Boston)]))
print(pred[1:5])

# compare it with the R built-in linear regression model
rlm = lm(medv ~ ., data=Boston)
print(rlm$coef)
print(rlm$fitted[1:5])
